### PR TITLE
misc: cleanup - const-ness and more use of abseil string utilities

### PR DESCRIPTION
**Pull request by derekargueta**
Description:
* mark variables `const` as appropriate
* mark methods as `const` as appropriate
* migrate `fmt::format("... {}", foo);` to `absl::StrCat("... ", foo);`
* migrate some `const std::string&` function parameter types to `absl::string_view` where appropriate

Risk Level: low
Testing: existing
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Comment by antoniovicente** [comment]
This would have been easier to review if const changes were done in a different PR than absl::StrCat cleanups.

File: source/common/common/perf_annotation.cc [hunk 1]
```diff
@@ -108,7 +108,7 @@ std::string PerfAnnotationContext::toString() {
             : std::to_string(
                   std::chrono::duration_cast<std::chrono::nanoseconds>(stats.total_).count() /
                   count));
-    columns[3].push_back(fmt::format("{}", stats.stddev_.computeStandardDeviation()));
+    columns[3].push_back(absl::StrCat("", stats.stddev_.computeStandardDeviation()));
```
Comment (antoniovicente) [inline source/common/common/perf_annotation.cc]: "" as the first argument to absl::StrCat seems unnecessary.

**Comment by derekargueta** [comment]
@antoniovicente sorry this branch creeped in scope.

File: source/common/common/perf_annotation.cc (hunk 1 above)
Comment (derekargueta) [reply source/common/common/perf_annotation.cc]: you're right

**Pull request merged**